### DISCUSSION TITLE
Welcome to ts-sql-query Discussions!

### DISCUSSION THREAD

**Discussion by juanluispaz** [juanluispaz/ts-sql-query#5] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by xialvjun** [comment]
Can you publish new version in npm? So we can test it.
Since `ts-sql-query` is still in its v0, quickly changing api is acceptable.

**Reply by juanluispaz** [reply]
I'm planning to release the first 1.0 version.

I already committed al breaking changes that I wanted to grow up in features in the future; they are small.

The biggest one affects the connection and table definition. Now you must include a unique name and nothing else as a generic argument.
 
I think it is a good moment to publish the 1.0 version with a stable API.

I will release the new version later today.

**Reply by xialvjun** [reply]
You can release it as `0.18.0`, and after a few weeks of using by people like us, then release it as `1.0`

**Reply by juanluispaz** [reply]
Released as `1.0.0-beta.1`

**Comment by xialvjun** [comment]
Now the doc is split into many chunks -- Yes, it's just one markdown, but things indeed is split separate...

New bee like me need a easy example like:
```ts
// db.ts
import { Pool } from 'pg';
import { Table } from 'ts-sql-query/Table';
import { PostgreSqlConnection } from 'ts-sql-query/connections/PostgreSqlConnection';
import { PgPoolQueryRunner } from 'ts-sql-query/queryRunners/PgPoolQueryRunner';

class DBConection extends PostgreSqlConnection<DBConection, 'DBConnection'> {}

const pool = new Pool({
  connectionString: 'postgresql://127.0.0.1:5432/sql_turotial',
});

export const countries = new (class extends Table<DBConection> {
  country_id = this.primaryKey('country_id', 'string');
  country_name = this.optionalColumn('country_name', 'string');
  region_id = this.column('region_id', 'int');
  constructor() {
    super('countries'); // table name in the database
  }
})();

export const regions = new (class extends Table<DBConection> {
  region_id = this.autogeneratedPrimaryKey('region_id', 'int');
  region_name = this.optionalColumn('region_name', 'string');
  constructor() {
    super('regions'); // table name in the database
  }
})();

export const conn = new DBConection(new PgPoolQueryRunner(pool));

// index.ts
import * as db from './db';

(async () => {
  const countries = await db.conn
    .selectFrom(db.countries)
    .select({ id: db.countries.country_id, name: db.countries.country_name })
    .executeSelectMany();
  console.log(countries);
  process.exit();
})();
```

This example is for `v0.17.0`.

**Reply by juanluispaz** [reply]
I've added a new section in the readme that link to the examples already included in the source code.

**Comment by juanluispaz** [comment]
I just published a new beta release with version `1.0.0-beta.1` that included all the changes implemented until now. See the [release notes](https://github.com/juanluispaz/ts-sql-query/releases/tag/v1.0.0-beta.1) for more details.
